bokeh/document.py

```diff
@@ -386,7 +417,7 @@ def remove_on_change(self, *callbacks):
 
         '''
         for callback in callbacks:
-            self._callbacks.remove(callback)
+            self._callbacks.pop(callback)
```

Comment:
`del self._callbacks[callback]` feels more natural if we aren't going to use the value